### PR TITLE
add changelog

### PR DESCRIPTION
**Pull request by aleeusgr**
# Description
Intersect MBO Open Source Office governance policy requires that CHANGELOG.md exists at the top level of the project.
The log should link to individual logs for the components. #5874 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by carbolymer — CHANGES_REQUESTED** [review]
It does not render readably: https://github.com/IntersectMBO/cardano-node/blob/9bdf7ab901bd4f20a0b937175aac73f801d5b893/CHANGELOG.md . Would you mind using markdown links to link the sub-changelogs?

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## 0.1.0.0 -- August 2024
```
Comment (carbolymer) [inline CHANGELOG.md:1]: This version number is incorrect

**Comment by aleeusgr** [comment]
Hi! @carbolymer 

I added the changes you requested: removed version number and added hyperlinks respective .md's;

**Review by carbolymer — COMMENTED** [review]
LGTM, but can you squash the commits together?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Changelog
+
+Changelogs for components can be found as follows:
+
+[trace-resources](https://github.com/IntersectMBO/cardano-node/blob/master/trace-resources/CHANGELOG.md) <br>
```
Comment (disassembler) [inline CHANGELOG.md:5]: Use bullets instead of <br> at end of line

**Review by disassembler — CHANGES_REQUESTED** [review]

**Pull request closed without merging**